src/math/FGFunction.cpp

```diff
@@ -802,12 +803,14 @@ FGFunction::~FGFunction()
 
 bool FGFunction::IsConstant(void) const
 {
-  for (auto p: Parameters) {
-    if (!p->IsConstant())
-      return false;
-  }
-
-  return true;
+  if (constForConstParams) {
+    for (auto p: Parameters) {
+      if (!p->IsConstant())
+        return false;
+    }
+    return true;
+  } else
+    return false;
```

Comment:
This seems too complicated to me. Since all the functions with no parameters are meant to be non-const why not simply test if Parameters is empty ?
```diff
bool FGFunction::IsConstant(void) const {
+  if (Parameters.empty()) return false;

  for (auto p: Parameters) {
    if (!p->IsConstant())
      return false;
  }

  return true;
}
```

Reply:
Yep, I did mention that as an option in the issue report a while ago, but hadn't received any feedback on which approach was the better option.

> or modify FGFunction::IsConstant() to return false if there are no parameters, i.e. assume functions with no parameters aren't constant.

So I went with the more flexible approach, e.g. maybe we add a function that doesn't take any arguments but it does return a constant value. Although there is no real harm in terms of an incorrect result, just that we could potentially lose some optimization if this constant function was expensive to run. But that is a lot of ifs....

But I guess there is `YAGNI` to consider :wink:

Reply:
> Yep, I did mention that as an option in the issue report a while ago, but hadn't received any feedback on which approach was the better option.

To be honest, I did not read all the details of the conversation, what I grasped was that the functions '<switch>` and `<urandom>` were incorrectly considered constant by the code and that this should be fixed.

> But I guess there is `YAGNI` to consider :wink:

I could not tell any better :smile: